src/azure-cli/azure/cli/__main__.py

```diff
@@ -72,35 +72,46 @@ def cli_main(cli, args):
 
     try:
         # check for new version auto-upgrade
-        if az_cli.config.getboolean('auto-upgrade', 'enable', False) and \
+        if exit_code == 0 and az_cli.config.getboolean('auto-upgrade', 'enable', False) and \
                 sys.argv[1] != 'upgrade' and (sys.argv[1] != 'extension' or sys.argv[2] != 'update'):
             from azure.cli.core._session import VERSIONS  # pylint: disable=ungrouped-imports
             from azure.cli.core.util import get_cached_latest_versions, _VERSION_UPDATE_TIME  # pylint: disable=ungrouped-imports
             if VERSIONS[_VERSION_UPDATE_TIME]:
                 import datetime
                 version_update_time = datetime.datetime.strptime(VERSIONS[_VERSION_UPDATE_TIME], '%Y-%m-%d %H:%M:%S.%f')
-                if datetime.datetime.now() > version_update_time + datetime.timedelta(days=10):
+                if datetime.datetime.now() > version_update_time + datetime.timedelta(days=11):
```

Comment:
> (days=11)

May I ask why the interval of update time need be changed?

Reply:
Considering our release cycle is 21 days and 2 checks of 10 days may miss a new version while 2 checks of 11 days would guanrantee a discovery of new versions.

Reply:
Got it